lib/cfg-lex.l

```diff
@@ -301,7 +301,7 @@ word	[^ \#'"\(\)\{\}\[\]\\;\r\n\t,|\.@:]
                              else
                                g_string_append(yyextra->string_buffer, yytext);
                            }
-<INITIAL,filterx>{white}+  ;
+<INITIAL,filterx>{white}+  { g_string_truncate(yyextra->token_text, 0); }
```

Comment:
@bazsi This patch changes the output or `--preprocess-into`. I think it's worth keeping `--preprocess-into` as intact as possible.

Reply:
You are right, good catch.

I started seeing to spaces added to my tokens, but this seems to be an issue with the YY_BREAK macro:

```
#define YY_BREAK \
  do {                                                                  \
    if (yyextra->token_text && YY_START == INITIAL)                     \
      {                                                                 \
        g_string_append_len(yyextra->token_pretext, yyextra->token_text->str, yyextra->token_text->len); \
        g_string_truncate(yyextra->token_text, 0);                      \
      }                                                                 \
  } while (0);                                                          \
  break;
```

As you can see this would strip the whitespace from the token, but only in case the YY_START is INITIAL.

Which is not true for the filterx state, which operates in a new lexer state.

I think the solution is:

```
diff --git a/lib/cfg-lex.l b/lib/cfg-lex.l
index 457cafe52..054bf1303 100644
--- a/lib/cfg-lex.l
+++ b/lib/cfg-lex.l
@@ -171,7 +171,7 @@ yy_filter_input(CfgLexer *self, gchar *buf, gsize buf_size)
 
 #define YY_BREAK \
   do {                                                                  \
-    if (yyextra->token_text && YY_START == INITIAL)                     \
+    if (yyextra->token_text && (YY_START == INITIAL || YY_START == filterx)) \
       {                                                                 \
         g_string_append_len(yyextra->token_pretext, yyextra->token_text->str, yyextra->token_text->len); \
         g_string_truncate(yyextra->token_text, 0);                      \
@@ -301,7 +301,7 @@ word        [^ \#'"\(\)\{\}\[\]\\;\r\n\t,|\.@:]
                              else
                                g_string_append(yyextra->string_buffer, yytext);
                            }
-<INITIAL,filterx>{white}+  { g_string_truncate(yyextra->token_text, 0); }
+<INITIAL,filterx>{white}+  ;
 <INITIAL,filterx>\.\.      { return LL_DOTDOT; }
 <INITIAL,filterx>\.\.\.    { return LL_DOTDOTDOT; }
 <INITIAL,filterx>=>        { return LL_ARROW; }
@@ -344,7 +344,7 @@ word        [^ \#'"\(\)\{\}\[\]\\;\r\n\t,|\.@:]
     /* and this is the version in filterx, which only allows "traditional" identifiers */
 <filterx>[a-zA-Z][_a-zA-Z0-9]*            { return cfg_lexer_map_word_to_token(yyextra, yylval, yylloc, yytext); }
 <filterx>\$\{({word}+(\.)?)*{word}+\}   { yylval->cptr = strndup(yytext+2, strlen(yytext) - 3); return LL_MESSAGE_REF; }
-<INITIAL>\,               { g_string_truncate(yyextra->token_text, 0); }
+<INITIAL>\,               ;
 
 <INITIAL,filterx>\"        {
                             g_string_truncate(yyextra->string_buffer, 0);
```

Reply:
I have edited the patch and pushed into your branch to update this PR. let's hope @kira-syslogng will be fixed with this.